test/runtests.jl

```diff
@@ -395,6 +404,15 @@ end
         ds = string(@doc DocumentedV1)
         @test contains(ds, "Very detailed documentation")
     end
+
+    @testset "parameterized" begin
+        @test typeof(ParamV1(i=1)) === ParamV1{Int}
+        @test typeof(ParamV1{Integer}(i=1)) === ParamV1{Integer}
+        @test typeof(ParamV1{Int}(i=1.0)) === ParamV1{Int}
+        @test_throws TypeError ParamV1{Float64}(i=1)
+        @test_throws TypeError ParamV1(i=1.0)
+        @test_throws InexactError ParamV1{Int}(i=1.1)
```

Comment:
these all need `;` for the kwarg to be YASG compliant :eye-of-yas:

```suggestion
        @test typeof(ParamV1(; i=1)) === ParamV1{Int}
        @test typeof(ParamV1{Integer}(; i=1)) === ParamV1{Integer}
        @test typeof(ParamV1{Int}(; i=1.0)) === ParamV1{Int}
        @test_throws TypeError ParamV1{Float64}(; i=1)
        @test_throws TypeError ParamV1(; i=1.0)
        @test_throws InexactError ParamV1{Int}(; i=1.1)
```